service/share/rpc.go

```diff
@@ -0,0 +1,80 @@
+package share
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/celestiaorg/celestia-node/node/rpc"
+)
+
+const namespacedSharesEndpoint = "/namespaced_shares"
+
+// sharesByNamespaceRequest represents a `GetSharesByNamespace`
+// request payload
+type sharesByNamespaceRequest struct {
+	NamespaceID string `json:"namespace_id"`
+	Root        string `json:"root"`
+}
+
+func (s *service) RegisterEndpoints(rpc *rpc.Server) {
+	rpc.RegisterHandlerFunc(namespacedSharesEndpoint, s.handleSharesByNamespaceRequest, http.MethodGet)
+}
+
+func (s *service) handleSharesByNamespaceRequest(w http.ResponseWriter, r *http.Request) {
+	// parse payload from request
+	buf, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		log.Errorf("serving %s request: %s", namespacedSharesEndpoint, err.Error())
+		return
+	}
+	// unmarshal payload
+	var req sharesByNamespaceRequest
+	err = json.Unmarshal(buf, &req)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		log.Errorf("serving %s request: %s", namespacedSharesEndpoint, err.Error())
+		return
+	}
```

Comment:
nit: This can be simplified with `json.NewDecoder(r.Body).Decode(&req)`. Same for the other PR with PFD.

Reply:
resolved

Reply:
What about other RP with sending PFM? If it is merged, would be nice to make a subsequent PR to fix this as well. Should not take a lot of time

Reply:
Yes i will do it rn